Publishing the standard models

mjethanandani opened the pull request:
This pull request publishes the draft version of the models as standard models, following the approval of them. This should be accompanied by the published version of MEF 58, which I will do next.

Couple of things to review:
- Where should the json files for Swagger be published. Currently they are in the build directory.
- Currently the YANG models are built in the build directory, and need to be manually copied over to the src/model/standard directory. I have done that to prevent churn in the standard folder. The build directory gets updated everytime someone does a make.

Finally, as part of integrating the models with the Unimgr project in OpenDaylight, I ran into issues with the model. Surprisingly, none of the other tools caught them.

mjethanandani committed:
Publishing the standard model.

mjethanandani committed:
Adding the published version of MEF 58.

mjethanandani committed:
Integrating with OpenDaylight, found these bugs.

daviball commented:
The history and errata in the README.md need to be updated.

daviball commented:
Also the revision statements in the draft versions are incomplete, and ni the standard versions the MEF spec number is missing.

mjethanandani committed:
Merge remote-tracking branch 'MEF-GIT/master'

mjethanandani committed:
Addressed comments from review and other fixes

daviball commented on src/model/draft/common/mef-types.yang:
```diff
@@ -43,11 +43,18 @@ module mef-types {
 
   revision YYYY-MM-DD {
```
Date is missing

mjethanandani replied:
I mentioned in my comments that I need to "build" the model once the changes are approved. That "build" will add the date.

daviball replied:
Committed version needs to have proper revision statements, including the date, so that people can use these models correctly.

mjethanandani replied:
David, the way I am doing it allows us to review the changes in the **source** file first. Once approved, and before I commit this change set, I will do a build of all the models, at which time the proper revision statement will be added to the model, add those models with proper revision statement to the commit list. If you want to hold off on approving the entire PR, I am fine. Just approve the changes, and as a next step, I will add the models with the correct revision statement.

daviball replied:
So long as the versions in src/model/ have the right revision etc, I am fine.  These are the versions people are looking at.

daviball commented on src/model/draft/common/mef-types.yang:
```diff
@@ -43,11 +43,18 @@ module mef-types {
-      "Initial Version.";
+      "Revision to address bugs found in integration..";
     reference
       "Legato - Services YANG SCA (MEF XX)";
```
Spec number is missing

mjethanandani replied:
Will replace XX with 58 in the reference statement.

daviball commented on src/model/draft/mef-global.yang:
```diff
@@ -47,10 +47,17 @@ module mef-global {
      'Carrier Ethernet Management Information Model', February 2017";
 
   revision YYYY-MM-DD {
```
Date is missing

mjethanandani replied:
Same comment as above.

daviball commented on src/model/draft/mef-legato-interfaces.yang:
```diff
@@ -39,10 +39,17 @@ module mef-legato-interfaces {
      'Carrier Ethernet Management Information Model', February 2017";
 
   revision YYYY-MM-DD {
```
Date is missing

mjethanandani replied:
Same comment as above.

daviball commented on src/model/draft/mef-legato-interfaces.yang:
```diff
@@ -495,9 +502,12 @@ module mef-legato-interfaces {
                    Envelopes.";
                 leaf envelope-id {
                   type leafref {
-                    path "../../ingress-envelopes/envelope/id";
```
This leafref previously refered to envelopes on this UNI, but now refers to envelopes on all UNIs, which is wrong.  All the similar changes you have made have this problem.  Either go back to relative paths, or limit the UNI to the current UNI.

mjethanandani replied:
Ok. Let me check.

daviball commented on src/model/draft/mef-legato-services.yang:
```diff
@@ -46,10 +46,17 @@ module mef-legato-services {
       February 2015";
 
   revision YYYY-MM-DD {
```
Missing date

mjethanandani replied:
Same comment as above.

mjethanandani committed:
Restore path for envelope-id and other changes.

mjethanandani committed:
For some reason these files did not make it in the last commit.

mjethanandani commented:
I have restored the leafref to a relative path and addressed some of your comments. There is one place where I apparently forgot, but it is in the description section.

> On Mar 16, 2018, at 8:13 AM, daviball <notifications@github.com> wrote:
> 
> @daviball commented on this pull request.
> 
> In src/model/draft/mef-legato-interfaces.yang <https://github.com/MEF-GIT/YANG-public/pull/6#discussion_r175119094>:
> 
> > @@ -39,10 +39,17 @@ module mef-legato-interfaces {
>       'Carrier Ethernet Management Information Model', February 2017";
>  
>    revision YYYY-MM-DD {
> Date is missing
> 
> In src/model/draft/mef-legato-interfaces.yang <https://github.com/MEF-GIT/YANG-public/pull/6#discussion_r175121332>:
> 
> > @@ -495,9 +502,12 @@ module mef-legato-interfaces {
>                     Envelopes.";
>                  leaf envelope-id {
>                    type leafref {
> -                    path "../../ingress-envelopes/envelope/id";
> This leafref previously refered to envelopes on this UNI, but now refers to envelopes on all UNIs, which is wrong. All the similar changes you have made have this problem. Either go back to relative paths, or limit the UNI to the current UNI.
> 
> In src/model/draft/mef-legato-services.yang <https://github.com/MEF-GIT/YANG-public/pull/6#discussion_r175121486>:
> 
> > @@ -46,10 +46,17 @@ module mef-legato-services {
>        February 2015";
>  
>    revision YYYY-MM-DD {
> Missing date
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/MEF-GIT/YANG-public/pull/6#pullrequestreview-104601616>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AH4OyBBA5eGxHd7-YGNYMhhXPhhAaliCks5te9aIgaJpZM4Q8Y-x>.
> 

Mahesh Jethanandani
mjethanandani@gmail.com

mjethanandani committed:
One more change to address the path in the description statement.

daviball commented on src/model/draft/mef-legato-services.yang:
```diff
@@ -914,7 +921,9 @@ module mef-legato-services {
+                          path "/mef-services/carrier-ethernet/" +
+	                       "subscriber-services/evc/cos-names/" +
+                               "cos-name/name";
                         }
```
This is still broken; it is now refering to CoS Names in all EVCs instead of CoS names in this EVC.  Please revert to the relative path.  Same applies in all the cases below.  Please revert them all to relative paths.

mjethanandani committed:
Reverted some of the paths back to relative paths

mjethanandani committed:
Missed one change

mjethanandani committed:
Adding build files.
